utils/service_discovery/abstract_config_store.py

```diff
@@ -208,8 +237,18 @@ def crawl_config_template(self):
             self.previous_config_index = config_index
             return False
         # Config has been modified since last crawl
+        # in this case a full config reload is triggered and the image_to_checks cache is rebuilt
         if config_index != self.previous_config_index:
-            log.info('Detected an update in config template, reloading check configs...')
+            log.info('Detected an update in config templates, reloading check configs...')
             self.previous_config_index = config_index
+            self.image_to_checks = self._populate_image_to_checks()
             return True
         return False
+
+    def _update_image_to_checks(self, image, check_names):
+        """Try to insert in the image_to_checks cache the mapping between an image and its check names"""
+        if image not in self.image_to_checks:
+            self.image_to_checks[image] = set(check_names)
+        elif self.image_to_checks[image] != set(check_names):
+            log.warning("Trying to cache check names for image %s but a different value is already there. "
+                        "This should not happen. Not updating." % image)
```

Comment:
Well none should happen, but we can cope silently with the first one. However for the second one there's no way to know which value to keep. Do you think we should add a warning for the other branch too?